db/migrate/20220615130617_create_metrics.rb

```diff
@@ -0,0 +1,39 @@
+class CreateMetrics < ActiveRecord::Migration[7.0]
+  def change
+    create_table :metrics do |t|
+      t.column :metric_action, :tinyint, null: false, default: 0
+      t.belongs_to :track, null: true
+      t.belongs_to :user, null: true
+
+      t.timestamps
+
+      t.index :created_at
+    end
+
+    %i[month day].each do |period|
+      create_table "metric_period_#{period}s".to_sym do |t|
+        t.column period, :tinyint, null: false, default: 0
+        t.column :metric_action, :tinyint, null: false, default: 0
+        t.belongs_to :track, null: true
+        t.integer :count, null: false
+
+        t.timestamps
+
+        t.index :created_at
+        t.index [period, :metric_action, :track_id], unique: true, name: 'uniq'
+      end
+    end
+
+    create_table :metric_period_minutes do |t|
+        t.integer :minute, limit: 2, null: false, default: 0
```

Comment:
I had to move this out of the above meta-programming bit as the minutes range from 0 to 1439, which doesn't fit a `tinyint`.